movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx

```diff
@@ -4,64 +4,103 @@ import { Localization } from '../../services/constants';
+  }
+  values.onCloseModal();
+};
 
-    if (!target.id.value) {
-      props.onAddMovie({
-        id,
-        title,
-        description,
-        fullDescription
-      });
-    } else {
-      props.onUpdateMovie({ id, title, description, fullDescription });
-    }
-    props.onCloseModal();
-  };
+function AddEditMovieForm(props: MovieFormProps & FormikProps<FormValues>): JSX.Element {
+  const { touched, errors, isSubmitting } = props;
+  const { t } = useTranslation();
 
   return (
     <>
       <h2>{t(Localization.ADD_FORM_TITLE)}</h2>
       <Button type={ButtonTypes.button} onClick={props.onCloseModal} title={t(Localization.LABEL_CLOSE)} />
-      <form id="modalForm" onSubmit={submitForm}>
+      <Form>
         <label>
           {t(Localization.LABEL_TITLE)}:
-          <input type="text" id={'title'} defaultValue={props.currentValue.title} />
+          <Field type="text" id={'title'} name={'title'} />
+          {touched.title && errors.title && <div>{errors.title}</div>}
         </label>
         <label>
           {t(Localization.LABEL_DESCRIPTION)}:
-          <input type="text" id={'description'} defaultValue={props.currentValue.description} />
+          <Field type="text" id={'description'} name={'description'} />
+          {touched.description && errors.description && <div>{errors.description}</div>}
         </label>
         <label>
           {t(Localization.LABEL_DESCRIPTION_FULL)}:
-          <input type="text" id={'fullDescription'} defaultValue={props.currentValue.fullDescription} />
+          <Field type="text" id={'fullDescription'} name={'fullDescription'} />
+          {touched.fullDescription && errors.fullDescription && <div>{errors.fullDescription}</div>}
         </label>
-        <input type="hidden" id={'id'} defaultValue={props.currentValue.id} />
-        <Button title={t(Localization.SUBMIT)} type={ButtonTypes.submit} />
-      </form>
+        <input type="hidden" id={'id'} />
+        <Button title={t(Localization.SUBMIT)} type={ButtonTypes.submit} disabled={isSubmitting} />
+      </Form>
     </>
   );
 }
+
+// The type of props MyForm receives
+interface WrapperFormProps extends MovieFormProps {
+  initialTitle?: string;
+  initialDescription?: string;
+  initialFullDescription?: string;
+}
+
+const WrapperForm = withFormik<WrapperFormProps, FormValues>({
+  mapPropsToValues: (props) => {
+    return {
+      ...props,
+      title: props.currentValue.title || '',
+      description: props.currentValue.description || '',
+      fullDescription: props.currentValue.fullDescription || ''
+    };
+  },
+
+  validate: (values: FormValues) => {
+    const errors: FormikErrors<FormValues> = {};
+    if (!values.title) {
+      errors.title = 'Required';
+    }
+    if (!values.description) {
+      errors.description = 'Required';
+    }
+    return errors;
+  },
+
+  handleSubmit: (values) => {
+    submitForm(values);
+  }
+})(AddEditMovieForm);
```

Comment:
hoc is not antipattern, but it's always hard to debug - better to have or additional regular parent component with appropriate props passing logic or hook approach.